packages/vda-node-manager/test/user_read.test.ts

```diff
@@ -330,29 +372,33 @@ describe('vda-node-manager read only tests', () => {
                 const regionCodes = [...new Set(regionArr)]
     
                 for (let i = 0; i < regionCodes.length; i++) {
-                    const result = await blockchainApi.getNodesByRegionCode(regionCodes[i], EnumStatus.active);
+                    const result = await blockchainApi.getNodesByRegionCode(regionCodes[i], 100, 1, EnumStatus.active);
                     assert.ok(result.length > 0, "Get 'active' nodes by region code");
                 }
 
                 // Get `remove started` nodes
                 const did = REMOVE_START_DIDS[0];
                 const removedNode = DID_NODE_MAP.get(did.address);
                 const removedregionCode = removedNode.regionCode;
-                const result = await blockchainApi.getNodesByRegionCode(removedregionCode, EnumStatus.removing);
+                let result = await blockchainApi.getNodesByRegionCode(removedregionCode, 100, 1, EnumStatus.removing);
                 assert.ok(result.length > 0, "Get 'removing' nodes by region code and status");
+
+                // Check pagination
+                result = await blockchainApi.getNodesByRegionCode(regionCodes[0], 1, 2, EnumStatus.active);
+                assert.ok(result.length === 1, "Get 'active' nodes by region code");
             })
 
             it("Get nodes by status", async () => {
```

Comment:
For all these pagination tests, can you also add a test to show the behaviour if you request results outside of the number of nodes in the contract.

ie: there are 100 nodes, but you request nodes from 150-180.

Also, what happens if you request 10,000 nodes? Is that too many? Do we need to limit the page size?

Reply:
1. When there are 100 nodes and you requested nodes from 150-180, it will returns an empty array.
2. We need to limit the page size

Reply:
Have you limited the page size?

Reply:
Pls check following functions:
```
getNodesByCountryCode()
getNodesByRegionCode()
getNodesByStatus()
```

Reply:
Pls check following files:
`vda-node-client/src/blockchain/blcochainApi.ts`
`vda-node-manager/src/blockchain/ownerApi.ts`